### PR TITLE
add `disjoint()` and `merge()`

### PR DESCRIPTION
**Pull request by jyn514**
I added lots of tests :)

### CODE REVIEW

File: codespan/src/span.rs [hunk 1]
```diff
@@ -44,21 +44,45 @@ impl Span {
...
+    /// assert!(span1.disjoint(span2));
+    /// ```
+    pub fn disjoint(self, other: Span) -> bool {
+        let (first, last) = if self.end < other.end {
+            (self, other)
+        } else {
+            (other, self)
+        };
+        first.end <= last.start
+    }
```
Comment (brendanzab) [inline codespan/src/span.rs]: nit:
```suggestion
    }

```

File: codespan/src/span.rs [hunk 2]
```diff
@@ -111,3 +135,62 @@ where
...
+        assert_eq!(two_four.merge(a), (1..5).into());
+
+        // disjoint
+        let ten_twenty = (10..20).into();
+        assert_eq!(a.merge(ten_twenty), (1..20).into());
+        assert_eq!(ten_twenty.merge(a), (1..20).into());
+
+        // identity
+        assert_eq!(a.merge(a), a);
+    }
```
Comment (brendanzab) [inline codespan/src/span.rs]: nit:
```suggestion
    }

```

**Comment by jyn514** [comment]
Addressed review comments.

**Comment by brendanzab** [comment]
Thanks!

**Pull request merged**